api/v1alpha1/shipwrightbuild_types_test.go

```diff
@@ -0,0 +1,57 @@
+package v1alpha1
+
+import (
+	"testing"
+
+	o "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ConditionReady    = "Ready"
+	ConditionNotReady = "NotReady"
+)
+
+// TestIsReady tests IsReady condition status function
+func TestIsReady(t *testing.T) {
+	g := o.NewGomegaWithT(t)
+	ready := ShipwrightBuildStatus{
+		Conditions: []metav1.Condition{
+			metav1.Condition{
+				Type:   ConditionReady,
+				Status: metav1.ConditionTrue,
+				Reason: "Good",
+			},
+			metav1.Condition{
+				Type:   ConditionNotReady,
+				Status: metav1.ConditionFalse,
+				Reason: "Good",
+			},
+		},
+	}
```

Comment:
This is more of a style thing for unit tests. Golang projects generally use table-driven tests in unit tests. The pattern is as follows:

1. Create a test cases slice with an in-lined struct:

```go
cases := []struct{
    name string
    buildStatus ShipwrightBuildStatus
    expectedResult bool
}{
    {
         name: "ready"
         expectedResult: true
         buildStatus: ...
    },
}
```

2. Loop through the cases, invoking a separate test run:

```go
for _, tc := range testCases {
    t.Run(tc.name, func(t *testing.T){
        ...
    }
}
```

In this test here, I would avoid using gomega matching since it's simpler to use the test case data:

```
   if status.IsReady() != tc.expectedResult {
      t.Errorf("expected isReady to be %t, got %t", tc.expectedResult, status.IsReady())
   }
```